pyscf/tools/trexio.py

```diff
@@ -438,3 +439,23 @@ def read_det_trexio(filename):
         det = trexio.read_determinant_list(tf, offset_file, num_det)
         return num_det, coeff, det
 
+def _to_segment_contraction(mol):
+    '''transform generally contracted basis to segment contracted basis
+    '''
+    _bas = []
+    for shell in mol._bas:
+        nctr = shell[gto.NCTR_OF]
+        if nctr == 1:
+            _bas.append(shell)
+            continue
+
+        nprim = shell[gto.NPRIM_OF]
+        pcoeff = shell[gto.PTR_COEFF]
+        bs = np.repeat(shell[np.newaxis], nctr, axis=0)
+        bs[:,gto.NCTR_OF] = 1
+        bs[:,gto.PTR_COEFF] = np.arange(pcoeff, pcoeff+nprim*nctr, nprim)
```

Comment:
Since this is a pointer into `_env`, shouldn't one also do something to `_env` to make this work?

Reply:
The contraction coefficients are stored in a way that the general contraction and segment contraction are compatible. It only requires an additional pointer to reference the individual shell within the contracted functions.